packages/core/src/components/icon/icon.tsx

```diff
@@ -73,9 +73,9 @@ export class Icon extends React.PureComponent<IIconProps & React.DOMAttributes<H
     public static readonly SIZE_STANDARD = 16;
     public static readonly SIZE_LARGE = 20;
 
-    public render() {
+    public render(): JSX.Element | null {
```

Comment:
seems like this should be caught by a test? there's some react 15 testing set up here right?

also how does this actually fix the problem? why is `Icon#render` inferred as `MaybeElement` in the first place...?

Reply:
`typeof icon !== "string"` includes booleans, which are not valid return values from `render()`.

we do have a react 15 setup but karma doesn't actually fail on compile errors (https://github.com/webpack-contrib/karma-webpack/issues/66)